climada/util/lines_polys_handler.py

```diff
@@ -0,0 +1,891 @@
+    at least two points.
+
+
+
+    Remark: LineString.interpolate() used here performs interpolation
+    on a geodesic.
+
+    Parameters
+    ----------
+    gdf_lines : gpd.GeoDataframe
+        Geodataframe with line geometries
+    res : float
+        Resolution (distance) in metres apart from which the generated Points
+        should be approximately placed.
+
+    Returns
+    -------
+    gdf_points : gpd.GeoDataFrame
+        Geodataframe with a double index, first for line geometries,
+        second for the point disaggregation of the lines (i.e. one Point
+        per row).
+
+    See also
+    --------
+    * util.coordinates.compute_geodesic_lengths()
+    """
+
+    # Needed because gdf.explode() requires numeric index
+    idx = gdf_lines.index.to_list() #To restore the naming of the index
+
+    gdf_points = gdf_lines.copy().reset_index(drop=True)
+    line_lengths = u_coord.compute_geodesic_lengths(gdf_points)
+
+    line_fractions = [
+        np.linspace(0, 1, num=_pnts_per_line(length, res))
+        for length in line_lengths
+        ]
+
+    gdf_points['geometry_pnt'] = [
+        shgeom.MultiPoint([
+            line.interpolate(dist, normalized=True)
+            for dist in fractions
+            ])
+        for line, fractions in zip(gdf_points.geometry, line_fractions)
+        ]
+
+    gdf_points = _swap_geom_cols(
+        gdf_points, geom_to='geometry_orig', new_geom='geometry_pnt'
+        )
+    gdf_points = gdf_points.explode()
+    gdf_points.index = gdf_points.index.set_levels(idx, level=0)
+    return gdf_points
+
+def _pnts_per_line(length, res):
+    return int(np.ceil(length / res) + 1)
+
+def invert_shapes(gdf_cutout, shape_outer):
+    """
+    given an outer shape and a gdf with geometries that are to be cut out,
+    return the remaining multipolygon
+    """
+    return shape_outer - _make_union(gdf_cutout)
+
+def _make_union(gdf):
+    """
+    Solve issue of invalid geometries in MultiPolygons, which prevents that
+    shapes can be combined into one unary union, save the respective Union
+    """
+
+    union1 = gdf[gdf.geometry.type == 'Polygon'].unary_union
+    union2 = gdf[gdf.geometry.type != 'Polygon'].geometry.buffer(0).unary_union
+    union_all = unary_union([union1, union2])
+
+    return union_all
+
+def _swap_geom_cols(gdf, geom_to, new_geom):
+    """
+    Change which column is the geometry column
+
+    Parameters
+    ----------
+    gdf : GeoDataFrame
+        Input geodatafram
+    geom_to : string
+        New name of the current 'geometry' column
+    new_geom : string
+        Column that should be set as the 'geometry' column. The column
+        new_geom is renamed to 'geometry'
+
+    Returns
+    -------
+    gdf_swap : GeoDataFrame
+        Copy of gdf with the new geometry column
+
+    """
+    gdf_swap = gdf.rename(columns = {'geometry': geom_to})
+    gdf_swap.rename(columns = {new_geom: 'geometry'}, inplace=True)
```

Comment:
```suggestion
    gdf_swap = gdf.rename(columns = {'geometry': geom_to,
    new_geom: 'geometry'}, inplace=True})
```

Reply:
Not sure, have you tried the tests? Should work I think.